scala-thoughts.md

```diff
@@ -0,0 +1,11 @@
+***What's Easy to do in Scala? What's hard?
+So far I think that I like pretty much everything about Scala. It really incorporates almost all of my favorite features from SML (a languge I initially hated but really really learned to love) while still having OOP and readable error messages (among other improvements). Obviously learning any language is hard and Scala is no exception, but despite my frustrations I really enjoyed my first spin. I think that Scala has a really really steep learning curve. Not an easy language to learn.
```

Comment:
The learning curve does feel really steep. I feel like I picked up C++, C, Python, Java, Python, Racket ... possibly even Prolog faster than I picked Scala up. But I guess that all of those languages are small compared to Scala, so it makes sense for the Scala learning curve to be even steeper (plus I'm sure the fog of time has made me forget my frustrations learning some of the other programming languages). 
I'm excited to learn how to do functional programming properly - I'm sure it'll serve me well in PLs (and maybe even the real world!).